src/main/java/io/confluent/idesidecar/restapi/featureflags/FeatureFlags.java

```diff
@@ -0,0 +1,738 @@
+      Duration scheduledInterval,
+      Consumer<FlagMutations<?>> defaultInitializer
+  ) {
+    // Configure the projects
+    this.deviceProjects = List.copyOf(deviceProjects);
+    this.ccloudProjects = List.copyOf(ccloudProjects);
+    this.allProjects = Stream
+        .concat(this.deviceProjects.stream(), this.ccloudProjects.stream())
+        .toList();
+    this.scheduledInterval = scheduledInterval != null ? scheduledInterval : Duration.ofSeconds(0);
+
+    // Build the device context and CCloud context (initially logged out)
+    updateDeviceContext();
+    updateCCloudContext(null);
+
+    // Initialize the defaults
+    if (defaultInitializer != null) {
+      defaultInitializer.accept(defaults());
+    }
+  }
+
+  /**
+   * Hook to be called after this bean starts up and after the {@link Inject @Injected} fields
+   * have been set.
+   *
+   * @param event the startup event
+   * @return the latch that can be used to wait for all projects' flags to be evaluated; never null
+   */
+  CountDownLatch startup(@Observes StartupEvent event) {
+    // Configure the contexts with the device information
+    updateDeviceContext();
+    updateCCloudContext(null);
+
+    // Evaluate the flags for the first time
+    return refreshFlags(true);
+  }
+
+  /**
+   * Get all the current flags.
+   *
+   * @return the stream of all flags for all projects
+   */
+  Map<String, FlagEvaluation> evaluationsById() {
+    Map<String, FlagEvaluation> combined = new HashMap<>(defaults.asMap());
+    allProjects
+        .stream()
+        .map(FeatureProject::evaluations)
+        .map(FlagEvaluations::asMap)
+        .forEach(combined::putAll);
+    // Then add the overrides
+    combined.putAll(overrides.asMap());
+    return combined;
+  }
+
+  /**
+   * Get the JSON representation of the first flag with the given ID in any of the projects.
+   *
+   * @param id the ID of the flag to find
+   * @return the current value of the flag, or empty if the flag does not exist
+   */
+  Optional<FlagEvaluation> findFirstFlag(@NotNull String id) {
+    // First check the overrides
+    var result = overrides.evaluation(id);
+    if (result.isEmpty()) {
+      // No luck, so look in the projects (in order)
+      result = allProjects
+          .stream()
+          .map(FeatureProject::evaluations)
+          .map(set -> set.evaluation(id).orElse(null))
+          .filter(Objects::nonNull)
+          .findFirst();
+    }
+    if (result.isEmpty()) {
+      // No luck, so look at the defaults
+      result = defaults.evaluation(id);
+    }
+    Log.debugf("Evaluating feature flag '%s' as: %s", id, result);
+    return result;
+  }
+
+  /**
+   * Get the JSON representation of the current value of the flag with the given ID
+   * from any of the projects.
+   *
+   * @param id the ID of the flag to find
+   * @return the current value of the flag, or empty if the flag does not exist
+   */
+  public Optional<JsonNode> evaluateAsJson(String id) {
```

Comment:
Good question. The main reason is that I originally intended `FlagId` to have literals for all of the feature flags **used in the sidecar**, and wanted to keep the REST API flexible by accepting any string ID. This way the VS Code extension could use flags that are not included in the `FlagId` enumeration.

I guess I still think this is the more flexible route, but I'm open to feedback. Regardless, I will update the JavaDoc on this method to better describe the difference between this method and the others that take a `FlagId` parameter rather than `String`.

Reply:
Thanks for the clarification!